cmd/zetaclientd/encrypt_tss.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var encTssCmd = &cobra.Command{
+	Use:   "tss-encrypt [file-path] [secret-key]",
+	Short: "Utility command to encrypt existing tss key-share file",
+	Args:  cobra.ExactArgs(2),
+	RunE:  EncryptTSSFile,
+}
+
+type TSSArgs struct {
+	secretKey string
+	filePath  string
+}
+
+var tssArgs = TSSArgs{}
+
+func init() {
+	RootCmd.AddCommand(encTssCmd)
+}
+
+func EncryptTSSFile(_ *cobra.Command, args []string) error {
+	tssArgs.filePath = args[0]
+	tssArgs.secretKey = args[1]
+
+	data, err := os.ReadFile(tssArgs.filePath)
+	if err != nil {
+		return err
+	}
+
+	if !json.Valid(data) {
+		return errors.New("file does not contain valid json, may already be encrypted")
+	}
+
+	block, err := aes.NewCipher(getFragmentSeed(tssArgs.secretKey))
+	if err != nil {
+		return err
+	}
+
+	// Creating GCM mode
+	gcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return err
+	}
+	// Generating random nonce
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return err
+	}
+
+	cipherText := gcm.Seal(nonce, nonce, data, nil)
+	return os.WriteFile(tssArgs.filePath, cipherText, 0o600)
```

Comment:
will this overwrite contents of existing file?

Reply:
Yes, if they want to do a backup it needs to be done before.